src/catalog/src/durable/impls/persist.rs

```diff
@@ -87,6 +86,43 @@ enum Mode {
     Writable,
 }
 
+/// Enum representing a potentially fenced epoch.
+#[derive(Debug)]
+enum PreOpenEpoch {
+    /// The current epoch CANNOT be fenced by a new epoch, it will just adopt any new epoch it sees.
+    /// Fencing can only occur after some data has been returned to a caller.
+    Unfenceable(Option<Epoch>),
+    /// The current epoch CAN be fenced by a new epoch.
+    Fenceable(Option<Epoch>),
+    /// The current epoch has been fenced.
+    Fenced {
+        current_epoch: Epoch,
+        fence_epoch: Epoch,
+    },
+}
```

Comment:
These semantics are a bit annoying but they match the behavior of the stash catalog. That behavior has probably developed organically without that much thought, but I'd rather match the stash behavior for now and revisit fencing once we start making the catalog shareable.